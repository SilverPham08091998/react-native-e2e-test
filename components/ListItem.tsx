import { Pressable, PressableProps, StyleProp, ViewStyle } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import Layout from 'constants/Layout';
import { useCallback } from 'react';
import { ThemeProps, useThemeColor } from './Themed';
import { Text } from './Text';

const base = {
  height: Layout.button,
  flexDirection: 'row',
  alignItems: 'center',
  paddingHorizontal: Layout.spacing.lg,
} as StyleProp<ViewStyle>;
export type ListItemProps = ThemeProps &
  PressableProps & {
    title: string;
    value?: boolean;
    onChangeValue?: (newValue: boolean) => void;
  };
export function ListItem(props: ListItemProps) {
  const { title, value, onChangeValue, lightColor, darkColor, style, ...otherPressableProps } =
    props;
  const name = value ? 'checkbox' : 'square-outline';
  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');

  const handleToggle = useCallback(() => {
    if (typeof onChangeValue === 'function') {
      onChangeValue(!value);
    }
  }, [onChangeValue, value]);

  return (
    <Pressable style={[base, style]} onPress={handleToggle} {...otherPressableProps}>
      <Text style={{ flexGrow: 1, flexShrink: 1 }}>{title}</Text>
      <Ionicons name={name} size={24} color={color} />
    </Pressable>
  );
}
