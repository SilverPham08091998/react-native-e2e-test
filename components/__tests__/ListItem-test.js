import * as React from 'react';
import renderer from 'react-test-renderer';

import { ListItem } from '../ListItem';

it(`renders correctly`, () => {
  const tree = renderer.create(<ListItem value={true} onChangeValue={newValue => console.log('newValue', newValue)} />).toJSON();

  expect(tree).toMatchSnapshot();
});
