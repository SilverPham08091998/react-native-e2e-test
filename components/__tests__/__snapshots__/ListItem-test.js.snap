// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
        "paddingHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 17,
          "fontWeight": "400",
          "letterSpacing": 0.3,
          "lineHeight": 20,
        },
        {
          "color": "#000",
        },
        {
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
  <Text />
</View>
`;
